clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsBuilderTest.java

```diff
@@ -72,19 +70,28 @@ public String toString() {
             List<Arguments> values = new ArrayList<>();
             for (int bufferOffset : Arrays.asList(0, 15))
                 for (CompressionType compressionType : CompressionType.values())
-                    values.add(Arguments.of(new Args(bufferOffset, compressionType)));
+                    if (accept(compressionType))
+                        values.add(Arguments.of(new Args(bufferOffset, compressionType)));
             return values.stream();
         }
+        protected boolean accept(CompressionType type) {
+            return true;
+        }
+    }
+
+    private static class NotZstd extends MemoryRecordsBuilderArgumentsProvider {
+        @Override
+        protected boolean accept(CompressionType type) {
+            return type != CompressionType.ZSTD;
+        }
     }
 
     private final Time time = Time.SYSTEM;
 
     @ParameterizedTest
-    @ArgumentsSource(MemoryRecordsBuilderArgumentsProvider.class)
+    @ArgumentsSource(NotZstd.class)
     public void testWriteEmptyRecordSet(Args args) {
         byte magic = RecordBatch.MAGIC_VALUE_V0;
-        assumeAtLeastV2OrNotZstd(magic, args.compressionType);
```

Comment:
It previously tested zstd with v2, but now it doesn't. Right?

Reply:
@ijuma  hello
The assume condition is `compressionType != CompressionType.ZSTD || magic >= MAGIC_VALUE_V2`
We specify the magic code which is less than `MAGIC_VALUE_V2` ,  `magic >= MAGIC_VALUE_V2` is always false .

So I think the condition is `compressionType != CompressionType.ZSTD  `

Reply:
Ah, yes, the magic is hardcoded here.